src/client/Microsoft.Identity.Client/Instance/AuthorityEndpoints.cs

```diff
@@ -15,5 +19,31 @@ public AuthorityEndpoints(string authorizationEndpoint, string tokenEndpoint, st
         public string AuthorizationEndpoint { get; }
         public string TokenEndpoint { get; }
         public string SelfSignedJwtAudience { get; }
+
+        public static async Task UpdateAuthorityEndpointsAsync(AuthenticationRequestParameters requestParameters)
```

Comment:
This code is common between RequestBase (i.e. calls to /token endpoint) and AuthCodeRequest (i.e. calls to /authorize), so I put it here.